docs/reference/inference/unified-inference.asciidoc

```diff
@@ -0,0 +1,412 @@
+[role="xpack"]
+[[unified-inference-api]]
+=== Unified inference API
+
+Streams a chat completion response using the Unified Schema format.
+
+IMPORTANT: The {infer} APIs enable you to use certain services, such as built-in {ml} models (ELSER, E5), models uploaded through Eland, Cohere, OpenAI, Azure, Google AI Studio, Google Vertex AI, Anthropic, Watsonx.ai, or Hugging Face.
+For built-in models and models uploaded through Eland, the {infer} APIs offer an alternative way to use and manage trained models.
+However, if you do not plan to use the {infer} APIs to use these models or if you want to use non-NLP models, use the <<ml-df-trained-models-apis>>.
+
+
+[discrete]
+[[unified-inference-api-request]]
+==== {api-request-title}
+
+`POST /_inference/<inference_id>/_unified`
+
+`POST /_inference/<task_type>/<inference_id>/_unified`
+
+
+[discrete]
+[[unified-inference-api-prereqs]]
+==== {api-prereq-title}
+
+* Requires the `monitor_inference` <<privileges-list-cluster,cluster privilege>>
+(the built-in `inference_admin` and `inference_user` roles grant this privilege)
+* You must use a client that supports streaming.
+
+
+[discrete]
+[[unified-inference-api-desc]]
+==== {api-description-title}
+
+The unified {infer} API enables real-time responses for completion tasks by delivering answers incrementally, reducing response times during computation.
+It only works with the `completion` task type for OpenAI and Elastic Inference Service. The Unified Schema defines a common schema for use across multiple services.
+
+
+[discrete]
+[[unified-inference-api-path-params]]
+==== {api-path-parms-title}
+
+`<inference_id>`::
+(Required, string)
+The unique identifier of the {infer} endpoint.
+
+
+`<task_type>`::
+(Optional, string)
+The type of {infer} task that the model performs. If included, this must be set to the value `completion`.
+
+
+[discrete]
+[[unified-inference-api-request-body]]
+==== {api-request-body-title}
+
+`messages`::
+(Required, array of objects) A list of objects representing the conversation.
```

Comment:
What do you think about adding some information about how these messages lists should be generated? Something like

"Requests should generally only add new messages from the user. The other messages ("assistent", "system", or "tool") should generally only be copy-pasted from the response to a previous completion request, such that the messages array is built up over the course of a conversation."